contracts/commands/ModernTriggerExecutor.sol

```diff
@@ -0,0 +1,183 @@
+
+    function getVerifier(uint16 triggerType) public view returns (IVerifier verifier) {
+        bytes32 validatorHash = keccak256(abi.encode("Verifier", triggerType));
+        verifier = IVerifier(serviceRegistry.getServiceAddress(validatorHash));
+    }
+
+    constructor(IServiceRegistry _serviceRegistry) {
+        if (address(_serviceRegistry) == address(0)) {
+            revert("modern-trigger/service-registry-not-registered");
+        }
+        bot = _serviceRegistry.getRegisteredService(AUTOMATION_BOT);
+        if (bot == address(0)) {
+            revert("modern-trigger/bot-not-registered");
+        }
+        operationExecutor = _serviceRegistry.getRegisteredService(OPERATION_EXECUTOR);
+        if (operationExecutor == address(0)) {
+            revert("modern-trigger/operation-executor-not-registered");
+        }
+        weth = _serviceRegistry.getRegisteredService(WETH);
+        if (weth == address(0)) {
+            revert("modern-trigger/weth-not-registered");
+        }
+        serviceRegistry = _serviceRegistry;
+        self = address(this);
+    }
+
+    /**
+     * @dev Validates the function selector of the execution data.
+     * @param expectedSelector The expected function selector.
+     * @param executionData The execution data containing the function selector.
+     * @dev Throws if the function selector is not equal to the expected selector.
+     */
+    function validateSelector(bytes4 expectedSelector, bytes memory executionData) public pure {
+        bytes4 selector = abi.decode(executionData, (bytes4));
+        require(selector == expectedSelector, "trigger/invalid-selector");
+    }
+
+    /**
+     * @notice Executes the trigger
+     * @param executionData Execution data from the Automation Worker
+     * @param triggerData  Encoded trigger data struct
+     */
+    function execute(
+        bytes calldata executionData,
+        bytes memory triggerData
+    ) external override nonReentrant {
+        require(bot == msg.sender, "modern-trigger/caller-not-bot");
+        ModernTriggerData memory modernTriggerData = abi.decode(triggerData, (ModernTriggerData));
+        IVerifier verifier = getVerifier(modernTriggerData.triggerType);
+        verifier.isTriggerTypeValid(modernTriggerData.triggerType);
+        ModernTriggerExecutor(self).validateoperationHash(
+            executionData,
+            modernTriggerData.operationHash
+        );
+        validateSelector(IOperationExecutor.executeOp.selector, executionData);
+        IAccountImplementation(modernTriggerData.positionAddress).execute(
+            operationExecutor,
+            executionData
+        );
+    }
+
+    /**
+     * @notice Returns the correctness of the vault state post execution of the command.
+     * @param triggerData Encoded trigger data struct
+     * @return Correctness of the trigger execution
+     */
+    function isExecutionCorrect(bytes memory triggerData) external view override returns (bool) {
```

Comment:
Similar comment to `checkContractEmpty`. I think is good that this function is external, but I wouldn't expect it to revert, and instead return the `bool` value. The caller will know what to do in that case

Reply:
can't do it right now as it's called by already deployed automation bot - so for now it has to stay as is.

Reply:
No problem them!